common/djangoapps/util/views.py

```diff
@@ -201,28 +197,19 @@ def reset_course_deadlines(request):
     Set the start_date of a schedule to today, which in turn will adjust due dates for
     sequentials belonging to a self paced course
     """
-    from lms.urls import RENDER_XBLOCK_NAME
-    from openedx.features.course_experience.urls import COURSE_HOME_VIEW_NAME
+    referrer = request.META.get('HTTP_REFERER')
+    if not referrer:
+        return
 
-    detail_id_dict = ast.literal_eval(request.POST.get('reset_deadlines_redirect_url_id_dict'))
-    redirect_url = request.POST.get('reset_deadlines_redirect_url_base', COURSE_HOME_VIEW_NAME)
-    course_key = CourseKey.from_string(detail_id_dict['course_id'])
-    masquerade_details, masquerade_user = setup_masquerade(
+    course_key = CourseKey.from_string(request.POST.get('course_id'))
+    _course_masquerade, user = setup_masquerade(
         request,
         course_key,
         has_access(request.user, 'staff', course_key)
     )
-    if masquerade_details and masquerade_details.role == 'student' and masquerade_details.user_name and (
-        redirect_url == COURSE_HOME_VIEW_NAME
-    ):
-        # Masquerading as a specific student, so reset that student's schedule
-        user = masquerade_user
-    else:
-        user = request.user
```

Comment:
`setup_masquerade` already does this logic for us and returns the right user.

Not sure why this logic was scoped to just the course home view, but that felt wrong, so I dropped that logic.